### PR TITLE
Update and clarify 1.14.5 release notes

### PR DESCRIPTION
**Pull request by patricklodder**
***IMPORTANT: I changed the reservations around BDB 5.3 upgrades***, as this was [confirmed to not cause any issues](https://github.com/dogecoin/dogecoin/pull/2483#issuecomment-903286757) and I personally have not experienced any issues on either Ubuntu (20.04, 21.04) and macOS since, having switched between wallets created with both versions multiple times. **If there is ANY doubt that there is a compatibility issue, then we MUST revert this change because this is a minor release.**

Updated the 1.14.5 release notes to:
- Include final code changes
- Explicitly mention that this now contains an important security update and recommend immediate update
- Clarify the new fee recommendation
  - Split wallet and relay policy changes
  - Link to the new recommendation document
  - Elaborate benefits and workarounds
  - Fix the wallet fee to be 0.01 DOGE, not 0.001.
- Reworded the version display in terms of user benefits instead of developer benefits.
- Shortened the section on key derivation path to be more concise, confident and just list the benefit.
- Reworded the section on the re-added auxpow methods, focusing on actual miner benefits
- Separated the RPC API changes as that's always a section of its own.
- Created a section about build system and CI, because those affect the useful life and thought it more useful to combine multiple improvements there
- Minor changes:
  - removed some of the individual commit changes that are now mentioned under the build system change
  - Added pruning test because in the past advice has been given that pruning doesn't work, and now it's proven to work
  - Reworded some of the changes to be more concise
- Fixed hard line endings to be more consistent with 80 characters (it was done in some places but not others)

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -2,70 +2,106 @@ Dogecoin Core version 1.14.5 is now available from:
...
 frequently tested on them.
 
 Notable changes
 ===============
 
+Security Fixes
+--------------
+
+This release addresses CVE-2021-3401 that opened potential remote code execution
+on QT wallets through malicious use of `dogecoin:` URIs.
```
Comment (rnicoll) [inline doc/release-notes.md:35]:
```suggestion
on QT (graphical user interface) wallets through malicious use of `dogecoin:` URIs.
```

**Review by rnicoll — APPROVED** [review]
One nit pick to clarify for non-technical users, then good to merge.

Realised I could check the BDB release notes, they confirm database file format has not changed:

* https://download.oracle.com/otndocs/products/berkeleydb/html/changelog_5_3.html#idp636088
* https://download.oracle.com/otndocs/products/berkeleydb/html/changelog_5_2.html#idp600136

Log format _has_ changed, in case we want to note it's important to ensure the node has shut down before upgrading, but I think we're okay to leave that.

File: doc/release-notes.md (hunk 1 above)
Comment (patricklodder) [reply doc/release-notes.md]: Thanks! done!

**Comment by patricklodder** [comment]
Drafting notes to include additional security fix for #2672.

**Review by edtubbs — CHANGES_REQUESTED** [review]

File: doc/release-notes.md [hunk 2]
```diff
@@ -2,70 +2,122 @@ Dogecoin Core version 1.14.5 is now available from:
...
-  standard transactions. The existing dust limit (-dustlimit) is now
-  enforcing the economic disincentive under which each output under
-  the limit must add additional fee to be accepted to the mempool.
-* Introduce `-discardthreshold`, a wallet-specific, configurable dust
-  limit that enables gradual implementation of the dust limit. Each
+  calculating fees, which significantly simplifies fee calculation logic and
+  makes it more similar to Bitcoin and Litecoin.
+* The default minimum transaction fee is now 0.01 DOGE per kilobyte. Note that
+  you may see transactions take longer to be confirmed while using these lower
+  fees, until all miners have updated. The new fee slider in the can help with
```
[the remark this answers is not in the document]
Comment (rnicoll) [reply doc/release-notes.md:81]:
```suggestion
  fees, until all miners have updated. The new fee slider can help with
```

[the remark this answers is not in the document]
Comment (patricklodder) [reply doc/release-notes.md]: Done!

**Comment by patricklodder** [comment]
Updated to fix @edtubbs' comment and squash-rebased.

**Review by rnicoll — APPROVED** [review]

**Pull request merged**